src/main/src/test/java/org/geoserver/config/GeoServerDataDirectoryTest.java

```diff
@@ -45,6 +45,7 @@ public static void beforeClass() {
 
     @AfterClass
     public static void afterClass() {
+        GeoServerExtensionsHelper.init(null);
```

Comment:
So what effect does that have? Do we need to update the javadoc to explain?

Reply:
This removes the ApplicationContext from GeoServerExtensions when this test class is finished so that the next test class that calls GeoServerExtensionsHelper.singleton won't modify the old ApplicationContext which will cause the stack trace every time GeoServerExtensions tries to look up a bean after that.  

This is done in several existing unit tests like:
https://github.com/geoserver/geoserver/blob/main/src/gwc/src/test/java/org/geoserver/gwc/GWCTest.java#L1646